src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/validators.py

```diff
@@ -322,7 +326,11 @@ def validate_authentication_type(namespace, formula=None):
         if namespace.admin_password:
             raise ValueError('Admin password cannot be used with SSH authentication type')
 
-        validate_ssh_key(namespace)
+        # Respect user's provided saved secret name for ssh authentication
+        if namespace.saved_secret:
+            namespace.ssh_key = "[[{}]]".format(namespace.saved_secret)
+        else:
+            validate_ssh_key(namespace)
```

Comment:
We need to check and issue an error if --ssh-key and --saved-secret are used together. This is a general CLI principle that we don't override or ignore conflicting parameters. It leads to customer confusion.